src/csharp/Grpc.IntegrationTesting/ServicesGrpc.cs

```diff
@@ -47,12 +47,28 @@ public interface IBenchmarkServiceClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IBenchmarkService
     {
       Task<global::Grpc.Testing.SimpleResponse> UnaryCall(global::Grpc.Testing.SimpleRequest request, ServerCallContext context);
       Task StreamingCall(IAsyncStreamReader<global::Grpc.Testing.SimpleRequest> requestStream, IServerStreamWriter<global::Grpc.Testing.SimpleResponse> responseStream, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class BenchmarkServiceBase
```

Comment:
To make this even more complicated, the name of the abstract class cannot be just Health because it is enclosed in an "umbrella" static class called "Health" (and member names cannot be same as their enclosing types).

So basically have these options:
1. leave HealthBase as is 
2. rename the umbrella class from "Health" to "HeathGrpc". I think that would actually look good in the code (you'd do things like `class HealthImpl : HealthGrpc.Health {}`  and `HealthGrpc.BindService(...)`). The problem is that all the code using the grpc stubs would need to be updated.  We might be able to afford this if we actually removed the client-side interfaces and figured out the ClientBase at the same time - because these would be breaking changes anyway.
3. come up with some other name of the class than "Health" or "HealthBase".